akka-sample-cqrs-scala/src/main/scala/sample/cqrs/ShoppingCartServer.scala

```diff
@@ -9,17 +9,16 @@ import akka.actor.typed.ActorSystem
 import akka.http.scaladsl.Http
 import akka.http.scaladsl.server.Route
 import akka.Done
-import akka.{ actor => classic }
 
-class ShoppingCartServer(routes: Route, port: Int, system: ActorSystem[_]) {
-  import akka.actor.typed.scaladsl.adapter._
-  implicit val classicSystem: classic.ActorSystem = system.toClassic
-  private val shutdown = CoordinatedShutdown(classicSystem)
+class ShoppingCartServer(routes: Route, port: Int)(implicit system: ActorSystem[_]) {
+  private val shutdown = CoordinatedShutdown(system)
 
   import system.executionContext
 
   def start(): Unit = {
-    Http().bindAndHandle(routes, "localhost", port).onComplete {
+    Http().newServerAt("localhost", port).bind(routes)
+      .map(_.addToCoordinatedShutdown(3.seconds))
```

Comment:
is the below `shutdown.addTask` still needed?

Reply:
something similar in `akka-sample-cqrs-java` ?

Reply:
It is not needed anymore.
Updated.